flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1325,26 +1497,38 @@ def run_to_convergence(self, desired_datasets, max_iterations=20):
             one desired state to another if there are no bugs.
         """
         for _ in range(max_iterations):
-            self.run_convergence_step(
-                dataset_map_from_iterable(desired_datasets))
-            note(self.current_datasets())
-            if compare_dataset_states(
-                self.current_datasets(),
-                dataset_map_from_iterable(desired_datasets),
-            ):
-                break
+            try:
+                self.run_convergence_step(
+                    dataset_map_from_iterable(desired_datasets))
+                if compare_dataset_states(
+                    self.current_datasets(),
+                    dataset_map_from_iterable(desired_datasets),
+                ):
+                    break
+            except Exception as e:
```

Comment:
On the one hand, yes, that's what real thing does, on the other hand I'm worried about real bugs being masked... Although I guess end-to-endness means convergence is what we really care about? OK I guess.